rest/config.go

```diff
@@ -1358,6 +1358,10 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 		for bucket, _ := range sc.Config.BucketCredentials {
 			buckets = append(buckets, bucket)
 		}
+		err = sc.BootstrapContext.Connection.SetConnectionStringServerless()
```

Comment:
I had thought the connection string should be getting set when BootstrapContext.Connection is initialized (in CreateCouchbaseClusterFromStartupConfig), and should only be using Serverless values when the SG node is running in serverless mode.

Reply:
When running through on the dubugger there was times when we hit the function below FetchDatabases and the connection string was simple just couchbase://localhost with no serverless connection options so that initial cluster connection was using the normal buffer. This was my way around that. I saw your comment about adding this in NewClusterConnection but I wasn't sure where I would get a serverless flag from there to ensure we only set it when runnning in serverless. Here there is already a check for if we are running in serverless so thought it was a good place to call it.

Reply:
I don't think I agree - I still feel like it would be better to properly initialize the values when the BootstrapContext is created, especially since you're setting it as a global, BootstrapContext-wide value, and it doesn't just apply to the next GetConfig call.  Seems like it would be easy for a future scenario to try to specify the value on the context, and have a fetchDatabase call unexpectedly override that value.

If there's times the connection string is just simple couchbase://localhost, I think the right thing to do is work out where that's being set in those cases.